Sources/NIO/BaseSocketChannel.swift

```diff
@@ -187,15 +368,16 @@ class BaseSocketChannel<T: BaseSocket>: SelectableChannel, ChannelCore {
         self.selectableEventLoop = eventLoop
         self.closePromise = eventLoop.newPromise()
         self.parent = parent
-        self.active.store(false)
         self.recvAllocator = recvAllocator
-        self._pipeline = ChannelPipeline(channel: self)
+        self.lifecycleManager = SocketChannelLifecycleManager(eventLoop: self.eventLoop,
+                                                              channelPipeline: ChannelPipeline(channel: self),
+                                                              isActiveAtomic: Atomic(value: false))
```

Comment:
We have a reference cycle here, right? `lifecycleManager` holds a reference to `self`, meaning `self` holds a cyclic reference to `self`.

Reply:
@Lukasa no because `ChannelPipeline` holds `Channel` as an `unowned` reference. It's totally non-obvious but has always been like this.